src/test/java/htsjdk/variant/vcf/VCFHeaderLineTranslatorUnitTest.java

```diff
@@ -14,49 +18,147 @@ public void testParseVCF4HeaderLine() {
+                                                                              null);
         Assert.assertEquals(values5.size(), 2);
         Assert.assertEquals(values5.get("ID"), "ANNOTATION");
         Assert.assertEquals(values5.get("Description"), "ANNOTATION \\n with a newline in it");
 
         // test with an unclosed quote
         try {
-            final Map<String, String> values6 = VCFHeaderLineTranslator.parseLine(VCFHeaderVersion.VCF4_2, "<ID=ANNOTATION,Description=\"ANNOTATION \\n with a newline in it>", null);
+            final Map<String, String> values6 = VCFHeaderLineTranslator.parseLine(VCFHeaderVersion.VCF4_2,
+                                                                                  "<ID=ANNOTATION,Description=\"ANNOTATION \\n with a newline in it>",
+                                                                                  null);
             Assert.fail("Should have thrown a TribbleException for having an unclosed quote in the description line");
-        } catch (TribbleException.InvalidHeader e) {
+        }
+        catch (TribbleException.InvalidHeader e) {
         }
 
         // test with an escaped quote at the end
         try {
-            final Map<String, String> values7 = VCFHeaderLineTranslator.parseLine(VCFHeaderVersion.VCF4_2, "<ID=ANNOTATION,Description=\"ANNOTATION \\n with a newline in it\\\">", null);
+            final Map<String, String> values7 = VCFHeaderLineTranslator.parseLine(VCFHeaderVersion.VCF4_2,
+                                                                                  "<ID=ANNOTATION,Description=\"ANNOTATION \\n with a newline in it\\\">",
+                                                                                  null);
             Assert.fail("Should have thrown a TribbleException for having an unclosed quote in the description line");
-        } catch (TribbleException.InvalidHeader e) {
         }
+        catch (TribbleException.InvalidHeader e) {
+        }
+
+    }
+
+    @DataProvider(name = "testData")
+    private Object[][] getTestData() {
+        String line = "<ID=X,Description=\"Y\">";
+        return new Object[][]{
+                {line, "ID,Description", "", ""},
+                {line, "ID", "Description", ""},
+                {line, "ID", "Description,Optional2", ""},
+                {line, "", "Description,ID", ""},
+                {line, "Description,ID", "", "Tag ID in wrong order"},
+                {line, "Description", "ID", "Optional tag ID must be listed after all expected tags"},
+                {line, "ID,Desc", "", "Unexpected tag Description"},
+                {line, "ID", "Desc", "Unexpected tag Description"},
+                {line, "ID,Description,Extra", "", ""},
+                {"<>", "", "", ""},
+                {"<>", "", "ID,Description", ""},
+                {"<>", "ID", "", ""},
```

Comment:
The empty tag cases <> did pass previously. They would create an output with empty key=value pair. I considered THAT a bug, so I fixed that. You want an exception thrown in this situation?

Reply:
> It would be much cleaner to separate the passing and failing test cases into separate data providers and test methods

I offered that before, seems like you're on a same page here.